oak-commons/src/test/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtilsTest.java

```diff
@@ -64,4 +64,10 @@ public void iteratorToSet() {
 
         Assert.assertEquals(s, CollectionUtils.toSet(iterable.iterator()));
     }
+
+    @Test
+    public void arrayToSet() {
+        final Set<String> s = CollectionUtils.toSet(data);
+        Assert.assertEquals(s, CollectionUtils.toSet(data.toArray()));
```

Comment:
That test case will will call the same method twice... If we change it to return an empty set, then the test won't fail

Reply:
Add a test case for the `null` element.

This throws a surprise result.
If we pass `null` to `toSet(T... elements)`, it gives a compilation error due to an ambiguous method call since we have 2 other methods with the same name.

On passing `null` after type casting to `Object` as `toSet((Object)null)`, it returns a `set` with one `null` element.

`@Test
    public void nullToSet() {
        Assert.assertEquals(new HashSet<>(), CollectionUtils.toSet((Object) null));
    }`

The above fails whereas (below) if we typecast it to `Object[]` passes. 
It returns the correct value of an empty set.

`@Test
    public void nullToSet() {
        Assert.assertEquals(new HashSet<>(), CollectionUtils.toSet((Object[]) null));
    }`

Reply:
is an empty set really desirable? Shouldn't we throw an NPE in both cases?